library/src/main/java/org/codice/security/sign/SimpleSign.java

```diff
@@ -241,6 +252,76 @@ public boolean validateSignature(
     }
   }
 
+  public void validateSignature(Signature signature) throws SignatureException {
+    RequestData requestData = new RequestData();
+    requestData.setSigVerCrypto(crypto.getSignatureCrypto());
+    WSSConfig wssConfig = WSSConfig.getNewInstance();
+    requestData.setWssConfig(wssConfig);
+
+    SAMLKeyInfo samlKeyInfo = null;
+
+    KeyInfo keyInfo = signature.getKeyInfo();
+    if (keyInfo != null) {
+      try {
+        samlKeyInfo =
+            SAMLUtil.getCredentialFromKeyInfo(
+                keyInfo.getDOM(),
+                new WSSSAMLKeyInfoProcessor(requestData),
+                crypto.getSignatureCrypto());
+      } catch (WSSecurityException e) {
+        throw new SignatureException("Unable to get KeyInfo.", e);
+      }
+    }
+    if (samlKeyInfo == null) {
+      throw new SignatureException("No KeyInfo supplied in the signature");
+    }
+
+    validateSignatureAndSamlKey(signature, samlKeyInfo);
+
+    Credential trustCredential = new Credential();
+    trustCredential.setPublicKey(samlKeyInfo.getPublicKey());
+    trustCredential.setCertificates(samlKeyInfo.getCerts());
+    Validator signatureValidator = new SignatureTrustValidator();
+
+    try {
+      signatureValidator.validate(trustCredential, requestData);
+    } catch (WSSecurityException e) {
+      throw new SignatureException("Error validating signature", e);
+    }
+  }
+
+  private void validateSignatureAndSamlKey(Signature signature, SAMLKeyInfo samlKeyInfo)
+      throws SignatureException {
+    SAMLSignatureProfileValidator validator = new SAMLSignatureProfileValidator();
+    try {
+      validator.validate(signature);
+    } catch (org.opensaml.xmlsec.signature.support.SignatureException e) {
+      throw new SignatureException("Error validating the SAMLKey signature", e);
+    }
+
+    BasicX509Credential credential = null;
+    if (samlKeyInfo.getCerts() != null) {
+      credential = new BasicX509Credential(samlKeyInfo.getCerts()[0]);
+    } else {
+      throw new SignatureException("Can't get X509Certificate or PublicKey to verify signature.");
+    }
+
+    ClassLoader threadLoader = null;
+    try {
+      threadLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
❓ Not really a review question, but why does DDF change the threads `contextClassLoader` here? Just want to know in case it's important or completely unnecessary.

Reply:
Not Sure SimpleSign is taken from DDF.